test/Hamiltonians.jl

```diff
@@ -837,6 +838,85 @@ end
     # offdiagonals
     @test num_offdiagonals(G2RealCorrelator(0), add1) == 0
     @test num_offdiagonals(G2RealCorrelator(0), comp) == 0
+
+    # Test show method
+    d = 5
+    output = @capture_out print(G2RealCorrelator(d))
+    @test output == "G2RealCorrelator($d)"
+end
+
+@testset "SuperfluidCorrelator" begin
+    m = 6
+    n1 = 4
+    n2 = m
+    add1 = BoseFS((n1,0,0,0,0,0))
+    add2 = near_uniform(BoseFS{n2,m})
+
+    # localised state
+    @test diagonal_element(SuperfluidCorrelator(0), add1) == n1/m
+    @test diagonal_element(SuperfluidCorrelator(1), add1) == 0.
```

Comment:
```suggestion
    @test @inferred diagonal_element(SuperfluidCorrelator(0), add1) == n1/m
    @test @inferred diagonal_element(SuperfluidCorrelator(1), add1) == 0.
```
Will cause an error if the code was type-unstable. 
In principle is should be sufficient to test this once under development, but having it here doesn't cost much and will make sure that further changes to the code are tested for type stability. It also signals to the reader that type stability was tested.